### PR TITLE
Improve JSON invalid fields error messages.

### PR DESCRIPTION
**Pull request by gierlachg**
Improved JSON invalid fields error messages (addresses #2759 & #2758). Added missing/unified tests.

### CODE REVIEW

**Comment by gierlachg** [comment]
verify

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataJsonResolver.java [hunk 1]
```diff
@@ -55,13 +55,17 @@ public String supportedFormat() {
...
+        }
+
         Map<QueryPath, MappingField> userFieldsByPath = extractFields(userFields, isKey);
 
         Map<String, MappingField> fields = new LinkedHashMap<>();
         for (Entry<QueryPath, MappingField> entry : userFieldsByPath.entrySet()) {
             QueryPath path = entry.getKey();
             if (path.getPath() == null) {
-                throw QueryException.error("Invalid external name: " + path.toString());
+                throw QueryException.error("Cannot use the '" + path + "' field with Json serialization");
```
Comment (viliam-durina) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataJsonResolver.java:68]:
```suggestion
                throw QueryException.error("Cannot use the '" + path + "' field with JSON serialization");
```

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MetadataJsonResolver.java [hunk 2]
```diff
@@ -57,13 +57,17 @@ public String supportedFormat() {
...
+        }
+
         Map<QueryPath, MappingField> userFieldsByPath = extractFields(userFields, isKey);
 
         Map<String, MappingField> fields = new LinkedHashMap<>();
         for (Entry<QueryPath, MappingField> entry : userFieldsByPath.entrySet()) {
             QueryPath path = entry.getKey();
             if (path.getPath() == null) {
-                throw QueryException.error("Invalid external name: " + path.toString());
+                throw QueryException.error("Cannot use the '" + path + "' field with Json serialization");
```
Comment (viliam-durina) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MetadataJsonResolver.java:70]:
```suggestion
                throw QueryException.error("Cannot use the '" + path + "' field with JSON serialization");
```

**Review by viliam-durina — APPROVED** [review]

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed.
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   StreamKafkaPTest.when_topicDoesNotExist_then_partitionCountGreaterThanZero ? Timeout
[INFO] 
[ERROR] Tests run: 35, Failures: 0, Errors: 1, Skipped: 0
[INFO] 

[ERROR] There are test failures.
</pre>
</details>

**Comment by gierlachg** [comment]
verify

**Pull request merged**